go/upgrade/api/api.go

```diff
@@ -174,20 +157,16 @@ func (d Descriptor) ValidateBasic() error {
 func (d *Descriptor) EnsureCompatible() error {
 	switch d.Method {
 	case UpgradeMethodInternal:
-		// For Internal upgrade method, the current binary hash needs to match the
-		// descriptor identifier.
-		OwnHash, err := OwnHash()
-		if err != nil {
-			return fmt.Errorf("error obtaining own binary hash: %w", err)
-		}
+		ownVersion := version.Versions
 
-		var descriptorHash hash.Hash
-		if err = descriptorHash.UnmarshalHex(d.Identifier); err != nil {
-			return fmt.Errorf("can't decode stored upgrade identifier: %w", err)
+		var descriptorVersion version.ProtocolVersions
+		if err := cbor.Unmarshal(d.Identifier, &descriptorVersion); err != nil {
+			return fmt.Errorf("can't decode descriptor upgrade identifier: %w", err)
 		}
 
-		if !OwnHash.Equal(&descriptorHash) {
-			return fmt.Errorf("binary hash missmatch: own: %d, required: %d", OwnHash, descriptorHash)
+		// TODO: allow patch version difference?
```

Comment:
Yeah could ignore patch.

Reply:
Hm so i think we should either require exact match or ignore non-major. If we consider all such upgrades "breaking" changes then following the version scheme we would bump major version every-time in any case, so i think we can just match major versions? 

And yeah i think it's better to not do the exact version match as i imagine a scenario: upgrade is accepted and scheduled in X days, we need to do an urgent release before (e.g. to fix something critical). Hm now actually thinking about this, maybe we want to instead use the version as a `MinVersion` (e.g. the urgent fix could technically be breaking and we'd bump the major version as well)?

Reply:
I think matching only the major version makes sense.

Besides the version the upgrade in question must also have the upgrade handler that is defined by the upgrade, so I guess "min version" could also work as long as the upgrade handler is also there.

Reply:
Yea agreed I also think major version is fine, will update the pr

Reply:
done